src/MountedTraversal.js

```diff
@@ -177,6 +178,12 @@ export function parentsOfInst(inst, root) {
   return pathToNode(inst, root).reverse();
 }
 
+export function instMatchesObjectProps(inst, props) {
+  if (!isDOMComponent(inst)) return false;
+  const node = getNode(inst);
+  return !isEmpty(props) && isMatch(propsOfNode(node), props);
```

Comment:
the `!isEmpty` check seems redudant. On L318, you are ensuring that `props` can't be null. 

So it seems like the `isEmpty` call is just a perf concern to make sure users aren't doing `find({})`, but that should be the minority case. 

So you are actually just adding complexity for the typical case of `find({foo: bar})`.

Ultimately, I think the `isEmpty` call is superflous and negatively affecting performance in a very small way.